typescript/src/services/tbtc.ts

```
@@ -289,27 +290,9 @@ export class TBTC extends TBTCCore {
-            typeof starknetProvider.getChainId === "function"
-          ) {
-            // Provider-only - use placeholder address for backward compatibility
-            walletAddressHex = "0x0"
-          } else {
-            // Re-throw the error for invalid providers or invalid addresses
-            throw error
-          }
-        }
+        const walletAddressHex = await TBTC.extractStarkNetAddress(
+          starknetProvider
+        )
```

Biome (2.4.15)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)